operators/src/engine/operator.rs

```diff
@@ -62,99 +44,146 @@ pub trait VectorOperator: Operator + CloneableVectorOperator {
     }
 }
 
-/// Common methods for `RasterOperator`s
-#[typetag::serde(tag = "type")]
-pub trait RasterOperator: Operator + CloneableRasterOperator {
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct ExecutionContext;
+
+pub trait InitializedOperator {
+    type Descriptor: ResultDescriptor + Clone;
+    fn execution_context(&self) -> &ExecutionContext;
+
     /// Get the result type of the `Operator`
-    fn result_type(&self) -> RasterDataType;
+    fn result_descriptor(&self) -> Self::Descriptor;
 
-    /// Instantiate a `TypedRasterQueryProcessor` from a `RasterOperator`
-    fn raster_processor(&self) -> Result<TypedRasterQueryProcessor>;
+    /// Get the sources of the `Operator`
+    fn raster_sources(&self) -> &[Box<InitializedRasterOperator>];
+
+    /// Get the sources of the `Operator`
+    fn vector_sources(&self) -> &[Box<InitializedVectorOperator>];
+
+    /// Get the sources of the `Operator`
+    fn raster_sources_mut(&mut self) -> &mut [Box<InitializedRasterOperator>];
+
+    /// Get the sources of the `Operator`
+    fn vector_sources_mut(&mut self) -> &mut [Box<InitializedVectorOperator>];
+}
+
+pub type InitializedVectorOperator =
+    dyn InitializedOperatorB<VectorResultDescriptor, TypedVectorQueryProcessor>;
+
+pub type InitializedRasterOperator =
+    dyn InitializedOperatorB<RasterResultDescriptor, TypedRasterQueryProcessor>;
+
+pub trait InitializedOperatorB<R, Q>: InitializedOperator<Descriptor = R> + Send + Sync
```

Comment:
After careful review I came to the conclusion that merging the two traits, or converting the InitializedOperator trait to a concrete type is not really feasible if we want to avoid code duplication and keep the ability to store operator state in the generic way that is stated. I just renamed the traits instead b932b17cd5d802740f694c86fc110d7aab3b3ed0